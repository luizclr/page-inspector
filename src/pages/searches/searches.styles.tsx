import styled from "styled-components";

import { Card } from "~/components/card/card";

export const Title = styled.h2`
  font-size: 18pt;
  margin-bottom: 20px;
  margin-left: 30px;
  color: #474b4e;
`;

export const Wrapper = styled.div`
  display: flex;
`;

export const SearchListWrapper = styled.div`
  display: flex;
  flex-direction: column;
  border-right: 2px solid #c6c6c6;
  padding-right: 15px;
  max-height: 250px;
`;

export const SearchList = styled.ul`
  width: 200px;
  height: 100%;
  overflow-y: auto;
  padding-right: 10px;
`;

export const SearchListItem = styled(Card)`
  width: 100%;
  margin-bottom: 5px;
  padding: 10px;
`;

export const ReloadButton = styled.button`
  margin: 15px;
  font-size: 12pt;
  padding: 10px;
  border: none;
  border-radius: 5px;
  border: 1px solid #5dc460;
  background-color: #fff;
  color: #42ab49;
  font-weight: bold;

  :hover {
    background-color: #5dc460;
    color: #fff;
  }
`;

export const RemoveButton = styled.button`
  margin: 15px;
  font-size: 12pt;
  padding: 10px;
  border: none;
  border-radius: 5px;
  border: 1px solid #ff6961;
  background-color: #fff;
  color: #c63637;
  font-weight: bold;

  :hover {
    background-color: #ff6961;
    color: #fff;
  }
`;

export const ResultListWrapper = styled.div`
  width: 100%;
`;

export const ResultList = styled.ul`
  list-style-type: none;
  max-height: 200px;
  overflow-y: auto;
  padding-left: 15px;
`;

export const ResultListItem = styled.li`
  padding: 5px;
`;
