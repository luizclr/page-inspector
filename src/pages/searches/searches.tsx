import { useEffect, useState } from "react";
import { useLocation } from "react-router-dom";

import { CardShadowSize, CardStyle } from "~/components/card/types";
import { Inspection } from "~/entities/inspection";
import {
  ReloadButton,
  RemoveButton,
  ResultList,
  ResultListItem,
  ResultListWrapper,
  SearchList,
  SearchListItem,
  SearchListWrapper,
  Title,
  Wrapper,
} from "~/pages/searches/searches.styles";
import { useInspection } from "~/providers/inspection/inspection";
import { InspectionActionTypes } from "~/reducers/inspector/types";
import { isNil } from "~/utils";

export const Searches: React.FC = () => {
  const [selectedInspection, setSelectedInspection] = useState<Inspection | null>(null);
  const { inspections, inspectionService, inspectionDispatch } = useInspection();
  const { state } = useLocation();

  useEffect(() => {
    if (!isNil(state?.id)) {
      void getInspection(state.id);
    }
  }, []);

  const getInspection = async (id: string): Promise<void> => {
    try {
      inspectionDispatch({
        type: InspectionActionTypes.loading,
        payload: {
          isLoading: true,
        },
      });
      const { data } = await inspectionService.getInspection({ id });

      setSelectedInspection(data);
      inspectionDispatch({
        type: InspectionActionTypes.add,
        payload: {
          inspection: data,
        },
      });
    } catch (error) {
      inspectionDispatch({
        type: InspectionActionTypes.error,
      });
    }
  };

  return (
    <div>
      <Title>Searches</Title>
      <Wrapper>
        <SearchListWrapper>
          <p>saved:</p>
          <SearchList>
            {inspections.map((el) => (
              <SearchListItem
                onClick={() => {
                  setSelectedInspection(el);
                }}
                key={el.id}
                cardStyle={CardStyle.dark}
                shadowSize={CardShadowSize.sm}
              >
                {el.id}
              </SearchListItem>
            ))}
          </SearchList>
        </SearchListWrapper>
        <ResultListWrapper>
          <div>
            <RemoveButton>remover</RemoveButton>
            <ReloadButton>reload</ReloadButton>
          </div>
          <ResultList>
            {selectedInspection?.urls.map((el) => (
              <ResultListItem key={el}>
                <a target="_blank" href={el} rel="noreferrer">
                  {el}
                </a>
              </ResultListItem>
            ))}
          </ResultList>
        </ResultListWrapper>
      </Wrapper>
    </div>
  );
};
