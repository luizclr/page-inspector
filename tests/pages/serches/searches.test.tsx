import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import { Searches } from "~/pages/searches/searches";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    state: { id: "id" },
  }),
}));

describe("<Searches />", () => {
  it("should be able to mount <Searches /> component", async () => {
    // given
    const { asFragment } = render(
      <BrowserRouter>
        <Searches />
      </BrowserRouter>
    );

    // when
    const sut = asFragment();

    // then
    expect(sut).toMatchSnapshot();
  });
});
