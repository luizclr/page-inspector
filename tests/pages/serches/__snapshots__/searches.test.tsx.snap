// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Searches /> should be able to mount <Searches /> component 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="sc-gswNZR YqaVI"
    >
      Searches
    </h2>
    <div
      class="sc-dkrFOg kJEZRB"
    >
      <div
        class="sc-hLBbgP jpKNMp"
      >
        <p>
          saved:
        </p>
        <ul
          class="sc-eDvSVe dhMdAi"
        />
      </div>
      <div
        class="sc-ftTHYK jctzJl"
      >
        <div>
          <button
            class="sc-iBYQkv gpdlHd"
          >
            remover
          </button>
          <button
            class="sc-gKPRtg cFTPBS"
          >
            reload
          </button>
        </div>
        <ul
          class="sc-pyfCe izPdum"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
